From stable-bounces@linux.kernel.org Tue Aug 22 17:17:47 2006
Message-Id: <200608230017.k7N0H5em003657@shell0.pdx.osdl.net>
From: Richard Purdie <rpurdie@rpsys.net>
From: akpm@osdl.org
Date: Tue, 22 Aug 2006 17:17:05 -0700
Cc: rpurdie@rpsys.net, stable@kernel.org, linux@dominikbrodowski.net
Subject: spectrum_cs: Fix firmware uploading errors

From: Richard Purdie <rpurdie@rpsys.net>

This fixes firmware upload failures which prevent the driver from working.

Signed-off-by: Richard Purdie <rpurdie@rpsys.net>
Cc: Dominik Brodowski <linux@dominikbrodowski.net>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/net/wireless/spectrum_cs.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.17.11.orig/drivers/net/wireless/spectrum_cs.c
+++ linux-2.6.17.11/drivers/net/wireless/spectrum_cs.c
@@ -245,7 +245,7 @@ spectrum_reset(struct pcmcia_device *lin
 	u_int save_cor;
 
 	/* Doing it if hardware is gone is guaranteed crash */
-	if (pcmcia_dev_present(link))
+	if (!pcmcia_dev_present(link))
 		return -ENODEV;
 
 	/* Save original COR value */
